csrc/scheduler/greedy.cpp

```diff
@@ -199,6 +238,92 @@ class CompileTimeChecker : private IterVisitor {
     }
   }
 
+  void handle(ScatterOp* scatter) override {
+    auto inp = scatter->in()->as<TensorView>();
+    auto out = scatter->out()->as<TensorView>();
+
+    if (out->getLogicalDomain().size() != 1) {
+      can_schedule_ = false;
+      setRejectReason(
+          "Scatter with multi-dimensional tensors is not yet supported");
+      return;
+    }
+
+    // Scatter input tensor is only allowed to be used by this scatter
+    // op itself due to the input-output aliasing
+    if (inp->uses().size() != 1) {
+      can_schedule_ = false;
+      setRejectReason(
+          "Scatter input can only be used by the scatter op itself");
+      return;
+    }
+
+    // Allocation domain must match between the input and output since
+    // they need to share the same memory buffer
+    if (inp->hasAllocation() && out->hasAllocation() &&
+        (exact_graph_.toGroups(inp->getAllocationDomain()) !=
+         exact_graph_.toGroups(out->getAllocationDomain()))) {
```

Comment:
Naive question: do we actually care about having explicit allocation? I'm wondering if it could be simplified to

```suggestion
    if (exact_graph_.toGroups(inp->getMaybeAllocationDomain()) !=
         exact_graph_.toGroups(out->getMaybeAllocationDomain()))) {
```

Reply:
I feel having `hasAllocation` is more intuitive since we are interested in the explicit allocation domain. Added this comment:

```
+    // If allocation domains already exist for the input and
+    // output, they must match since the input and output need to
+    // share the same memory buffer. This condition does not matter if
+    // no explicit allocation domain exists since we can set the
+    // allocation domain as required for this scheduler.
```